x-pack/plugins/watcher/server/models/watch_status_model/watch_status_model.test.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+
+import { WatchStatusUpstreamJson } from '../../../common/types';
+import { buildServerWatchStatusModel, buildClientWatchStatusModel } from './watch_status_model';
+
+const upstreamJson: WatchStatusUpstreamJson = {
+  id: 'my-watch',
+  watchStatusJson: {
+    version: 1,
+    state: {
+      active: true,
+      timestamp: '2017-03-02T14:25:31.139Z',
+    },
+    last_checked: '2017-03-02T14:25:31.139Z',
+    last_met_condition: '2017-07-05T14:25:31.139Z',
+    actions: {
+      foo: {
+        ack: {
+          timestamp: '2015-05-26T18:21:08.630Z',
+          state: 'awaits_successful_execution',
+        },
+        last_execution: {
+          timestamp: '2017-07-05T00:00:00.000Z',
+          successful: true,
+        },
+      },
+      bar: {
+        ack: {
+          timestamp: '2015-05-26T18:21:08.630Z',
+          state: 'awaits_successful_execution',
+        },
+        last_execution: {
+          timestamp: '2017-07-05T00:00:00.000Z',
+          successful: true,
+        },
+      },
+    },
+  },
+};
+
+describe('WatchStatusModel', () => {
+  describe('buildServerWatchStatusModel', () => {
+    it(`throws an error if no 'id' property in json`, () => {
+      expect(() => {
+        // @ts-ignore
+        buildServerWatchStatusModel({});
+      }).toThrow(/must contain an id property/i);
+    });
+
+    it(`throws an error if no 'watchStatusJson' property in json`, () => {
```

Comment:
Should we add a note that this test is temporary until all consumers are converted to TS?